assets/js/util/test/numberFormat.js

```diff
@@ -115,5 +116,105 @@ describe( 'numberFormat', () => {
 	it.each( siteKitLocales )( 'formats numbers correctly with locale variant %s', ( locale, value, expected ) => {
 		setupGoogleSiteKit( locale );
 		expect( numberFormat( value ) ).toStrictEqual( expected );
+		expect( console ).not.toHaveWarned();
+	} );
+
+	describe( 'graceful degradation for problematic options in some browsers', () => {
+		let NumberFormatSpy;
+
+		afterEach( () => {
+			if ( NumberFormatSpy.mockRestore ) {
+				NumberFormatSpy.mockRestore();
+			}
+		} );
+
+		// Error message that browser throws on error.
+		const errorMessage = 'TypeError: Failed to initialize NumberFormat since used feature is not supported in the linked ICU version';
+
+		/**
+		 * This function mocks the implementation of certain errors identified in
+		 * https://github.com/google/site-kit-wp/issues/3255
+		 * so that we can test that we've properly handled them.
+		 */
+		const mockNumberFormat = () => {
+			const NumberFormat = Intl.NumberFormat;
```

Comment:
Let's save this reference at the top module level where it will always reference the original. Otherwise it could end up grabbing a reference to the mocked implementation, depending on when it's called.

Reply:
Updated in https://github.com/google/site-kit-wp/pull/3620/commits/fd3d490c8d8864c2307c0a1e1296a8cfafd92cc0